### ISSUE TITLE
findConversation is liberal

### ISSUE THREAD

**Issue by johngriffin** [krismuniz/botkit-sms#7]
This is not necessarily a bug but it's something people need to consider.  

I used the findConversation method within my app in order to find and close existing conversations with users, before opening a new one.  However, I was just passing in a message containing  {text: "", channel: <phone_number>}.  In this case findConversation will always match the first conversation it finds because of the code:

`let matchesConvo = (
            convo.source_message.channel === message.channel ||
            convo.source_message.user === message.user
          )
`

As I was not using message.user anywhere it was always undefined.

**Comment by krismuniz** [comment]
Hmm... I'll think about how to solve this (if it has a solution). 

Probably just throw an error if `message.channel` & `message.user` are both `undefined`.

**Comment by johngriffin** [comment]
I think that ideally we would only use message.channel or message.user.  However I haven’t dug into the internals to see if this is feasible.

I only came across this issue because I was running startConversation(message, bot) with a message that I had constructed myself, only setting the message.channel.  I did not want this to be wrapped in a controller.hears() because I want the bot to initiate the conversation.

It may be enough to just advise in the README that if a user constructs their own message, they should set both message.user and message.channel?

**Comment by krismuniz** [comment]
Yes, I'll probably add that to the README as well. 

Still, I think that throwing an error is an appropriate thing to do. This immediate feedback could help a lot for solving this kind of simple problems.

**Comment by johngriffin** [comment]
Yes, there’s definitely something wrong if both message.channel and message.user are both undefined!

**Comment by krismuniz** [comment]
Exactly. :)
